src/BulletSoftBody/btConjugateGradient.h

```diff
@@ -0,0 +1,141 @@
+//
+//  btConjugateGradient.h
+//  BulletSoftBody
+//
+//  Created by Xuchen Han on 7/1/19.
+//
+
+#ifndef BT_CONJUGATE_GRADIENT_H
+#define BT_CONJUGATE_GRADIENT_H
+#include <iostream>
+#include <cmath>
+#include <LinearMath/btAlignedObjectArray.h>
+#include <LinearMath/btVector3.h>
+
+template <class TM>
+class btConjugateGradient
+{
+//    using TVStack = btAlignedObjectArray<btVector3>;
+    typedef btAlignedObjectArray<btVector3> TVStack;
+    TVStack r,p,z,temp;
+    int max_iterations;
+    
+public:
+    btConjugateGradient(const int max_it_in)
+    : max_iterations(max_it_in)
+    {
+    }
+    
+    virtual ~btConjugateGradient(){}
+    
+    // return the number of iterations taken
+    int solve(TM& A, TVStack& x, const TVStack& b, btScalar tolerance)
```

Comment:
TM is a bit too short of a name. Can you use something more similar to btMatrixX?